server/lorax_server/utils/sources/__init__.py

```diff
@@ -7,6 +11,21 @@
 HUB = "hub"
 S3 = "s3"
 LOCAL = "local"
+PBASE = "pbase"
+
+PREDIBASE_API_TOKEN = os.getenv("PREDIBASE_API_TOKEN", None)
+PREDIBASE_MODEL_URL_ENDPOINT = "/v1/models/version/name/{}?version={}"
+PREDIBASE_GATEWAY_ENDPOINT = os.getenv("PREDIBASE_GATEWAY_ENDPOINT", "https://api.predibase.com")
+
+@lru_cache(maxsize=256)
+def map_pbase_model_id_to_s3(model_id: str) -> str:
+    name, version = model_id.split("/")
+    url = PREDIBASE_GATEWAY_ENDPOINT + PREDIBASE_MODEL_URL_ENDPOINT.format(name, version)
+    headers = {"Authorization": f"Bearer {PREDIBASE_API_TOKEN}"}
```

Comment:
This works for dedicated deployments that are single-tenant, but would not work for serverless deployments, where each request could have a different API token. I think it's fine if we implement it like this for now, but we would need to revisit for serverless LLMs soon.

Reply:
yep have a todo to add the api_token as a header in the incoming request